app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
### [OPTIONAL]

- It will be great to use `[Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html)`  after_save to call your update_posts_counter method every time a new post has been saved to the database. This will be a much better implementation.